addons/l10n_pe_website_sale/static/src/js/website_sale.js

```diff
@@ -0,0 +1,104 @@
+/** @odoo-module **/
+import { WebsiteSale } from "@website_sale/js/website_sale";
+
+WebsiteSale.include({
+    events: Object.assign({}, WebsiteSale.prototype.events, {
+        "change select[name='state_id']": "_onChangeState",
+        "change select[name='city_id']": "_onChangeCity",
+    }),
+    start: function () {
+        this.selectCities = document.querySelector("select[name='city_id']");
+        this.selectDistricts = document.querySelector("select[name='l10n_pe_district']");
+        this.cityBlock = document.querySelector(".div_city");
+        this.autoFormat = document.querySelector(".checkout_autoformat");
+        this.selectState = document.querySelector("select[name='state_id']");
+        return this._super.apply(this, arguments);
+    },
+    _changeState: function () {
+        if (!this.selectState.value) {
+            return;
+        }
+        return this._rpc({
+            route: "/shop/state_infos/" + this.selectState.value,
+        }).then(data => {
+            // populate cities and display
+            if (data.cities?.length) {
+                this.selectCities.innerHTML = "";
+                data.cities.forEach(city => {
+                    let opt = document.createElement("option");
+                    opt.textContent = city[1];
+                    opt.value = city[0];
+                    opt.setAttribute("data-code", city[2]);
+                    this.selectCities.appendChild(opt);
+                });
+                this.selectCities.parentElement.style.display = "block";
+            } else {
+                this.selectCities.value = "";
+                this.selectCities.parentElement.style.display = "none";
+            }
+            this.selectDistricts.value = "";
+            this.selectDistricts.parentElement.style.display = "none";
+        });
+    },
+    _changeCity: function () {
```

Comment:
`_changeState` and `_changeCity` are the same function (except a few arguments). You can refactor it by extracting it to single function and call it in `_changeState` and `_changeCity`.

It will be easier to maintain.

Reply:
Got it, I ended up doing [this](https://github.com/odoo/odoo/blob/3c4286516f1d1aa5dab2589c3736335c12fc10d9/addons/l10n_pe_website_sale/static/src/js/website_sale.js#L17). What do you think? @rugo-odoo